packages/core/src/indices/BaseIndex.ts

```diff
@@ -91,10 +95,24 @@ export abstract class BaseIndex<T> {
    * Insert a document into the index.
    * @param document
    */
-  async insert(document: Document) {
+  async insert(
+    document: Document,
+    options?: { docStoreStrategy: DocStoreStrategy },
+  ) {
+    // if defined, run doc store strategy before other transformations
+    const docStoreStrategy = options?.docStoreStrategy
+      ? createDocStoreStrategy(
+          options.docStoreStrategy,
+          this.docStore,
+          this.vectorStore,
+        )
+      : undefined;
     const nodes = await runTransformations(
       [document],
-      [this.serviceContext.nodeParser],
+      [
+        ...(docStoreStrategy ? [docStoreStrategy] : []),
```

Comment:
```suggestion
        ...(docStoreStrategy && [docStoreStrategy])
```

I think this way is less prone to errors

and why it's an array? not the transformer directly?

Reply:
That's not doing the same thing. The idea is to prepend `docStoreStrategy` to the array literal only if it defined

Reply:
right, but the && operator will only pass if `docStoreStrategy ` is true, if not will not prepend, on the current way it prepend an empty array (which I think is unnecessary)

Reply:
oh that's right, the spread will let it empty